pkg/util/ipsec/ipsec_telemetry.go

```diff
@@ -0,0 +1,47 @@
+package ipsec
+
+import (
+	"strconv"
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+type ipsecAPIMode struct {
+	state     string
+	legacyAPI string // It is a string since we need a value for uninitailized state ("")
+}
+
+var (
+	registerMetrics sync.Once
+	ipsecStateGauge *metrics.GaugeVec
+
+	recordedMode ipsecAPIMode // initial values are empty strings
+)
+
+// RegisterIpsecMetrics in the global registry
+func RegisterIpsecMetrics() {
```

Comment:
Nit, replace Ipsec with IPsec, this is the pattern being used all over the places.

Reply:
Regarding the registration - this is not something I invented - some modules (e.g. pkg/cmd/checkendpoints/controller/metrics.go) use this method, but some (e.g. vendor/github.com/grpc-ecosystem/go-grpc-prometheus/client.go) use the init module, I can move it. Good catch on the name (should be IPsec)